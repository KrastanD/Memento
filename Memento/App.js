import "react-native-gesture-handler";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import EventsScreen from "./screens/EventsScreen";
import AddEventScreen from "./screens/AddEventScreen";

const Stack = createStackNavigator();

export default function App() {
  const [initialNavigationState, setInitialNavigationState] = React.useState();
  const containerRef = React.useRef();

  return (
    <NavigationContainer
      ref={containerRef}
      initialState={initialNavigationState}
    >
      <Stack.Navigator mode="modal">
        <Stack.Screen
          name="Events"
          component={EventsScreen}
          options={{
            title: "Memento",
            headerStyle: { backgroundColor: "#01a699" },
            headerTintColor: "#fff",
          }}
        />
        <Stack.Screen
          name="AddEvent"
          component={AddEventScreen}
          options={{
            headerStyle: { backgroundColor: "#01a699" },
            headerTintColor: "#fff",
          }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
