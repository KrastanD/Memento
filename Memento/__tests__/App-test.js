import React from "react";
import App from "../App";
import renderer from "react-test-renderer";

it("snapshots App screen correctly", () => {
  const tree = renderer.create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
});
