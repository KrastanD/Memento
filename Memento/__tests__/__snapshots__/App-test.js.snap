// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App screen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTScrollView
                        collapsable={false}
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#fafafa",
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              Object {
                                "flexDirection": "row",
                                "paddingHorizontal": 10,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <View>
                                <View
                                  pointerEvents="none"
                                  style={
                                    Array [
                                      Object {
                                        "bottom": 0,
                                        "left": 0,
                                        "position": "absolute",
                                        "right": 0,
                                        "top": 6,
                                      },
                                      Object {
                                        "backgroundColor": "#f6f6f6",
                                        "borderColor": "rgba(0, 0, 0, 0.54)",
                                        "borderRadius": 4,
                                        "borderWidth": 1,
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    Object {
                                      "minHeight": 64,
                                      "paddingBottom": 0,
                                      "paddingTop": 8,
                                    }
                                  }
                                >
                                  <View
                                    pointerEvents="none"
                                    style={
                                      Object {
                                        "bottom": 0,
                                        "left": 0,
                                        "opacity": 1,
                                        "position": "absolute",
                                        "right": 0,
                                        "top": 0,
                                        "transform": Array [
                                          Object {
                                            "translateX": 0,
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <View
                                      pointerEvents="none"
                                      style={
                                        Object {
                                          "backgroundColor": "#f6f6f6",
                                          "bottom": 0,
                                          "height": 2,
                                          "left": 10,
                                          "opacity": 0,
                                          "position": "absolute",
                                          "right": 0,
                                          "top": 6,
                                          "transform": Array [
                                            Object {
                                              "translateX": 0,
                                            },
                                          ],
                                          "width": 8,
                                        }
                                      }
                                    />
                                    <Text
                                      numberOfLines={1}
                                      style={
                                        Object {
                                          "backgroundColor": "#f6f6f6",
                                          "color": "transparent",
                                          "fontFamily": "System",
                                          "fontSize": 16,
                                          "fontWeight": undefined,
                                          "left": 18,
                                          "opacity": 0,
                                          "paddingHorizontal": 0,
                                          "position": "absolute",
                                          "textAlign": "left",
                                          "top": 37,
                                          "transform": Array [
                                            Object {
                                              "translateX": 0,
                                            },
                                            Object {
                                              "translateY": 0,
                                            },
                                            Object {
                                              "scale": 1,
                                            },
                                            Object {
                                              "scaleY": 1,
                                            },
                                          ],
                                          "writingDirection": "ltr",
                                        }
                                      }
                                    >
                                      Search
                                    </Text>
                                    <Text
                                      numberOfLines={1}
                                      onLayout={[Function]}
                                      style={
                                        Object {
                                          "color": "#01a699",
                                          "fontFamily": "System",
                                          "fontSize": 16,
                                          "fontWeight": undefined,
                                          "left": 0,
                                          "opacity": 0,
                                          "paddingHorizontal": 14,
                                          "position": "absolute",
                                          "textAlign": "left",
                                          "top": 36,
                                          "transform": Array [
                                            Object {
                                              "translateX": 0,
                                            },
                                            Object {
                                              "translateY": 0,
                                            },
                                            Object {
                                              "scale": 1,
                                            },
                                          ],
                                          "writingDirection": "ltr",
                                        }
                                      }
                                    >
                                      Search
                                    </Text>
                                    <Text
                                      numberOfLines={1}
                                      style={
                                        Object {
                                          "color": "rgba(0, 0, 0, 0.54)",
                                          "fontFamily": "System",
                                          "fontSize": 16,
                                          "fontWeight": undefined,
                                          "left": 0,
                                          "opacity": 1,
                                          "paddingHorizontal": 14,
                                          "position": "absolute",
                                          "textAlign": "left",
                                          "top": 36,
                                          "transform": Array [
                                            Object {
                                              "translateX": 0,
                                            },
                                            Object {
                                              "translateY": 0,
                                            },
                                            Object {
                                              "scale": 1,
                                            },
                                          ],
                                          "writingDirection": "ltr",
                                        }
                                      }
                                    >
                                      Search
                                    </Text>
                                  </View>
                                  <TextInput
                                    allowFontScaling={true}
                                    editable={true}
                                    multiline={false}
                                    onBlur={[Function]}
                                    onChangeText={[Function]}
                                    onFocus={[Function]}
                                    placeholder=""
                                    placeholderTextColor="rgba(0, 0, 0, 0.54)"
                                    rejectResponderTermination={true}
                                    selectionColor="#01a699"
                                    style={
                                      Array [
                                        Object {
                                          "flexGrow": 1,
                                          "margin": 0,
                                          "paddingHorizontal": 14,
                                          "textAlign": "left",
                                          "zIndex": 1,
                                        },
                                        Object {
                                          "height": 56,
                                        },
                                        Object {
                                          "paddingBottom": 0,
                                          "paddingTop": 0,
                                        },
                                        Object {
                                          "color": "#000000",
                                          "fontFamily": "System",
                                          "fontSize": 16,
                                          "fontWeight": undefined,
                                          "textAlignVertical": "center",
                                        },
                                      ]
                                    }
                                    underlineColorAndroid="transparent"
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#01a699",
                            "borderColor": "rgba(0,0,0,0.2)",
                            "borderRadius": 120,
                            "borderWidth": 1,
                            "bottom": 20,
                            "elevation": 2,
                            "height": 60,
                            "justifyContent": "center",
                            "opacity": 1,
                            "position": "absolute",
                            "right": 20,
                            "width": 60,
                          }
                        }
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                  <View
                    pointerEvents="box-none"
                  >
                    <View
                      accessibilityElementsHidden={false}
                      importantForAccessibility="auto"
                      onLayout={[Function]}
                      pointerEvents="box-none"
                      style={null}
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "#01a699",
                              "borderBottomColor": "rgb(224, 224, 224)",
                              "flex": 1,
                              "shadowColor": "rgb(224, 224, 224)",
                              "shadowOffset": Object {
                                "height": 0.5,
                                "width": 0,
                              },
                              "shadowOpacity": 0.85,
                              "shadowRadius": 0,
                            }
                          }
                        />
                      </View>
                      <View
                        pointerEvents="box-none"
                        style={
                          Object {
                            "height": 64,
                            "maxHeight": undefined,
                            "minHeight": undefined,
                            "opacity": undefined,
                            "transform": undefined,
                          }
                        }
                      >
                        <View
                          pointerEvents="none"
                          style={
                            Object {
                              "height": 20,
                            }
                          }
                        />
                        <View
                          pointerEvents="box-none"
                          style={
                            Object {
                              "alignItems": "center",
                              "flex": 1,
                              "flexDirection": "row",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Object {
                                "marginHorizontal": 18,
                              }
                            }
                          >
                            <Text
                              accessibilityRole="header"
                              numberOfLines={1}
                              onLayout={[Function]}
                              style={
                                Object {
                                  "color": "#fff",
                                  "fontSize": 17,
                                  "fontWeight": "600",
                                }
                              }
                            >
                              Memento
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
